providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -188,25 +187,41 @@ export class EthereumProvider implements IEthereumProvider {
   // ---------- Private ----------------------------------------------- //
 
   private registerEventListeners() {
-    this.signer.on("session_event", (payload: any) => {
+    this.signer.on("session_event", (payload: SignClientTypes.EventArguments["session_event"]) => {
       const { params } = payload;
       if (!this.rpc.chains.includes(params.chainId)) return;
       const { event } = params;
       if (event.name === "accountsChanged") {
         this.accounts = event.data;
         this.events.emit("accountsChanged", this.accounts);
       } else if (event.name === "chainChanged") {
-        this.setChainId(event.data);
+        this.setChainId(this.formatChainId(event.data));
       } else {
         this.events.emit(event.name, event.data);
       }
       this.events.emit("session_event", payload);
     });
 
-    this.signer.on("disconnect", () => {
-      this.events.emit("disconnect");
+    this.signer.on("chainChanged", (chainId: number) => {
+      this.chainId = chainId;
+      this.events.emit("chainChanged", chainId);
     });
 
+    this.signer.on(
+      "session_update",
+      (payload: SignClientTypes.EventArguments["session_update"]) => {
+        this.events.emit("session_update", payload);
+      },
+    );
+
+    this.signer.on(
+      "session_delete",
+      (payload: SignClientTypes.EventArguments["session_delete"]) => {
+        this.reset();
```

Comment:
does this need to also disconnect a "signer" similar to how `disconnect()` method does it?

Reply:
Nope, as the signer (`universal-provider`) does the clean-up before emitting the event